### DISCUSSION TITLE
Modeling problem

### DISCUSSION THREAD

**Discussion by EilanAlhersh** [metrumresearchgroup/mrgsolve#1090] [Q&A]
Hello, 

Thank you so much for your help. I'm new to PKPD modeling and I'm required to start directly with mrgsolve. 
I wrote the code but eacg time I'm trying to run the code, the modeling is failed to run.  would you please have a look to my code?

I have one more question: why were using demo in the mode?? I just saw many people are using demo! in Nonmum we use data file and we input the variable and run the model, can I know how we replace this step using mrgsolve? how should i include the covariate that i already have in CSV file! 

code2 <- '
$PARAM @annotated

V3: 191 : Peripheral volume (L)
CRCL : 80 : Creatinine clearance (mL/min)
WTKG : 70 : Weight (kg)

$CMT @annotated 

GUT : Gut compartment (mg)
CENT : Centrak compartment (mg)
PERI : Peripheral compartment (mg)

$MAIN

double TVKA = THETA(1);
double KA = TVKA * exp(ETA(1));

double TVCL = THETA(2) * (CRCL/72.950)**THETA(6) + THETA(8) * (WTKG/85.150);
double CL = TVCL * exp(ETA(2));

double TVV2 = THETA(3) * (WTKG/85.150)**THETA(7);
double V2 = TVV2 * exp(THETA(9)*ETA(2));

double TVV3 = THETA(4);
double V3 = TVV3;

double TVQ = THETA(5);
double Q = TVQ;

double K = CL/V2;
double K23 = Q/V2;
double K32 = Q/V3;

$ODE 
dxdt_GUT = -KA * GUT;
dxdt_CENT = KA * GUT - K * CENT - K23 * CENT + K32 * PERI;
dxdt_PERI = K23 * CENT - K32 * PERI;

$THETA @annotated

0, 2       : Absorption Rate Constant(Ka) (1/hr)
0, 15      : Apparent oral Clearance   (CL) (L/hr)
0, 200     : Apparent Volume of Distribution (V2) (L)
191 : Peripheral volume (V3) (L)
0, 20       : Intercompartmental clearance (Q) (L/hr)
0.001      : Change in log CL per unit change in log CRCL (L/hr)/(mL/min)
0.001      : Change in log V2 per unit change in log WTKG (L)/(kg)
0.001      : Change in CL per unit change in WTKG (L/hr)/(kg)
0.4        : Ratio between SDv2/SDcl

$OMEGA
0.2
0.07

$SIGMA 
1

$TABLE

capture DV = 1000 * CENT/V2;

$CAPTURE @annotated

DV : plasma concentration (ng/ml)

'

mod <- mcode_cache("demo", code2)

I really appreciate any kind of help

**Comment by kylebaron** [comment]
Hi @EilanAlhersh  - 

Here's a list of issues and I'll post the corrected code below

- Use `@annotated` rather than `@Annotated`
- To raise some base to a power, use the `pow()` function like this `pow(CRCL/72.950, THETA(6))`
- Remove `V3` from the first parameter block (you want that as a derived variable rather than parameter)

The code is 

```r
code2 <- '
$PARAM @annotated
CRCL : 80  : Creatinine clearance (mL/min)
WTKG : 70  : Weight (kg)

$CMT @annotated

GUT : Gut compartment (mg)
CENT : Centrak compartment (mg)
PERI : Peripheral compartment (mg)

$MAIN

double TVKA = THETA(1);
double KA = TVKA * exp(ETA(1));

double TVCL = THETA(2) * pow(CRCL/72.950, THETA(6)) + THETA(8) * (WTKG/85.150);
double CL = TVCL * exp(ETA(2));

double TVV2 = THETA(3) * pow(WTKG/85.150, THETA(7));
double V2 = TVV2 * exp(THETA(9)*ETA(2));

double TVV3 = THETA(4);
double V3 = TVV3;

double TVQ = THETA(5);
double Q = TVQ;

double K = CL/V2;
double K23 = Q/V2;
double K32 = Q/V3;

$ODE
dxdt_GUT = -KA * GUT;
dxdt_CENT = KA * GUT - K * CENT - K23 * CENT + K32 * PERI;
dxdt_PERI = K23 * CENT - K32 * PERI;

$THETA @annotated

0, 2 : Absorption Rate Constant(Ka) (1/hr)
0, 15 : Apparent oral Clearance (CL) (L/hr)
0, 200 : Apparent Volume of Distribution (V2) (L)
191 : Peripheral volume (V3) (L)
0, 20 : Intercompartmental clearance (Q) (L/hr)
0.001 : Change in log CL per unit change in log CRCL (L/hr)/(mL/min)
0.001 : Change in log V2 per unit change in log WTKG (L)/(kg)
0.001 : Change in CL per unit change in WTKG (L/hr)/(kg)
0.4 : Ratio between SDv2/SDcl

$OMEGA
0.2
0.07

$SIGMA
1

$TABLE

capture DV = 1000 * CENT/V2;

$CAPTURE @annotated

DV : plasma concentration (ng/ml)

'
```

**Comment by EilanAlhersh** [comment]
Thank you so so much :)

**Comment by EilanAlhersh** [comment]
if i have my own dataset. and i need to run the simulation 1000 times for each SubjectID. could we use nid?
